CMakeLists.txt

```diff
@@ -233,7 +245,7 @@ endif(NOT CPP_LIBRARY)
 #----- Sources
 
 # It does make no sense building it if there is no LLVM, given that the JIT implementation relies on LLVM.
-if(ENABLE_CLANG_AS_LIB)
+if(ENABLE_CLANG_AS_LIB AND ENABLE_JIT)
```

Comment:
I would prefer to automatically set `ENABLE_JIT` to `false` whenever there is no LLVM, and only use one flag here.

Reply:
makes sense to set `ENABLE_JIT` automatically to false if i don't have ClangAsLib. On the single flag issue: what about llvm 18 and 19 where I can have it but i don't have JitCompiler, as it has not been ported

Reply:
Clang as a library should be tested in the other test file, not here.
We can base this condition only on `ENABLE_JIT`.